### PR TITLE
use httpx.AsyncClient for async http requests

### PR DESCRIPTION
**Pull request by graingert**
requests.Session does blocking IO which blocks the event loop! No other requests can be processed concurrently

### CODE REVIEW

File: docs/first-server/python.mdx [hunk 1]
```diff
@@ -768,7 +790,7 @@ uvicorn.run(app, host="0.0.0.0", port=8000)
   </Step>
   <Step title="Run tests">
   ```bash
-  uv add --dev pytest pytest-asyncio
```
Comment (graingert) [inline docs/first-server/python.mdx:771]: httpx and starlette both depend on anyio which ships with an async test framework - so you don't need to install an extra one here

**Review by jspahrsummers — CHANGES_REQUESTED** [review]
The change to httpx looks great—thank you!

Re: `anyio`, although we use it in our SDK, it may make examples or getting started a bit more confusing than using `asyncio`, because the latter is part of the standard library. We've purposely opted to stick with `asyncio` for now. Do you mind reverting those bits? 🙏

**Comment by graingert** [comment]
Happy to drop the use of anyio.run, but having multiple async plugins for pytest installed is overkill

**Comment by dsp-ant** [comment]
> Happy to drop the use of anyio.run, but having multiple async plugins for pytest installed is overkill

To stay consistent across, the implementation should use one async runtime for now. I would think it's best if for now we keep asyncio and create an issue to follow up if we should just switch the whole example, including the implementation itself, to anyio given that it is transient a dependency anyway.

**Comment by graingert** [comment]
> > Happy to drop the use of anyio.run, but having multiple async plugins for pytest installed is overkill
> 
> To stay consistent across the example, the implementation should use one async runtime for now. I would think it's best if for now we keep asyncio and create an issue to follow up if we should just switch the whole example, including the implementation itself, to anyio given that it is transient a dependency anyway.

Yes I configured it to use asyncio only

**Review by jspahrsummers — APPROVED** [review]
Thank you!

**Pull request merged**